src/commonMain/kotlin/fr/acinq/lightning/db/PaymentsDb.kt

```diff
@@ -46,6 +46,15 @@ interface IncomingPaymentsDb {
 
     /** List received payments (with most recent payments first). */
     suspend fun listReceivedPayments(count: Int, skip: Int, filters: Set<PaymentTypeFilter> = setOf()): List<IncomingPayment>
+
+    /** List incoming payments (with the most recent payments first). */
+    suspend fun listIncomingPayments(count: Int, skip: Int, filters: Set<PaymentTypeFilter> = setOf()): List<IncomingPayment>
```

Comment:
I'm not sure we should add this function, I don't see it being used by wallets.
Users are only interested in listing the payments that they received.
@dpad85 let us know what you think.

Reply:
I added it to be extra careful in the test and check the exact state of the DB before and, more importantly, after purging invoices.

If it's only useful for testing so perhaps I should take it out. I can instead use the fact that only one incoming payment is purged, but not two, to ensure that partially paid payments aren't purged.

Reply:
Let's wait for dpad85 to decide!

Reply:
Indeed we don't need it in the app, but it's nice to have that check in the tests. So I think we should keep it, especially if `purgeExpiredPayments` does not return the affected rows count anymore (which affects the test).